src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Migrations/20.diff.sql

```diff
@@ -0,0 +1,353 @@
+       The partition anchor boundary will be removed at the very first run of the purge operation of the purge change data worker.
+       Total number of partition boundaries = the number of history partitions + one for the current hour + the number of future partitions. */
+    WHILE @numberOfHistoryPartitions >= -@numberOfFuturePartitions
+    BEGIN
+        /* Rounds the start datetime to the hour. */
+        SET @rightPartitionBoundary = DATEADD(hour, DATEDIFF(hour, 0, @currentDateTime) - @numberOfHistoryPartitions, 0);
+
+        /* Checks if a partition exists. */
+        IF NOT EXISTS (SELECT 1 FROM sys.partition_range_values AS prv
+                        JOIN sys.partition_functions AS pf
+                            ON pf.function_id = prv.function_id
+                    WHERE pf.name = N'PartitionFunction_ResourceChangeData_Timestamp'
+                        and CAST(prv.value AS datetime2(7)) = @rightPartitionBoundary) 
+        BEGIN
+            /* Creates new empty partition by creating new boundary value and specifying NEXT USED file group. */
+            ALTER PARTITION SCHEME PartitionScheme_ResourceChangeData_Timestamp NEXT USED [Primary];
+            ALTER PARTITION FUNCTION PartitionFunction_ResourceChangeData_Timestamp() SPLIT RANGE(@rightPartitionBoundary);
+        END;
+            
+        SET @numberOfHistoryPartitions -= 1;
+    END;
+END;
+GO
+
+EXEC dbo.LogSchemaMigrationProgress 'Creating a staging table for resource change data table.';
+GO
+
+/* Creates a staging table. */
+IF NOT EXISTS (SELECT 1 FROM sys.tables WHERE name = 'ResourceChangeDataStaging')
+BEGIN
+    /* Staging table that will be used for partition switch out. */
+    CREATE TABLE dbo.ResourceChangeDataStaging 
+    (
+        Id bigint IDENTITY(1,1) NOT NULL,
+        Timestamp datetime2(7) NOT NULL CONSTRAINT DF_ResourceChangeDataStaging_Timestamp DEFAULT sysutcdatetime(),
+        ResourceId varchar(64) NOT NULL,
+        ResourceTypeId smallint NOT NULL,
+        ResourceVersion int NOT NULL,
+        ResourceChangeTypeId tinyint NOT NULL,
+        CONSTRAINT PK_ResourceChangeDataStaging_TimestampId PRIMARY KEY (Timestamp, Id)
+    ) ON [PRIMARY]
+END;
+GO
+
+/* Adds a check constraint on the staging table for a partition boundary validation. */
+IF NOT EXISTS(SELECT 1 FROM sys.check_constraints WHERE name = 'CHK_ResourceChangeDataStaging_partition')
+BEGIN    
+    ALTER TABLE dbo.ResourceChangeDataStaging WITH CHECK 
+        ADD CONSTRAINT CHK_ResourceChangeDataStaging_partition CHECK(Timestamp < CONVERT(DATETIME2(7), N'9999-12-31 23:59:59.9999999'));
+END;
+GO
+
+IF EXISTS(SELECT 1 FROM sys.check_constraints WHERE name = 'CHK_ResourceChangeDataStaging_partition')
+BEGIN    
+    ALTER TABLE dbo.ResourceChangeDataStaging CHECK CONSTRAINT CHK_ResourceChangeDataStaging_partition;
+END;
+GO
+
+EXEC dbo.LogSchemaMigrationProgress 'Dropping the old primary key clustered index from resource change data table.';
```

Comment:
here as well - Deleting PK_ResourceChangeData

Reply:
Updated the log.